dali/daliadmin/daadmin.cpp

```diff
@@ -2264,6 +2264,98 @@ void getxref(const char *dst)
+        if (dFile)
+        {
+            bool locked = false;
+            COnScopeExit ensureUnlock([&]() { if (locked) dFile->unlockProperties(); });
+            if (dFile->querySuperFile())
+                WARNLOG("Skipping: file '%s' is a superfile", lfn);
+            else
+            {
+                unsigned numParts = dFile->numParts();
+                for (unsigned p=0; p<numParts; p++)
+                {
+                    IDistributedFilePart &part = dFile->queryPart(p);
+                    IPropertyTree &attrs = part.queryAttributes();
+                    if (!attrs.hasProp("@size"))
+                    {
+                        if (fix)
+                        {
+                            offset_t partSize;
+                            try
+                            {
+                                partSize = part.getFileSize(true, true);
+                            }
+                            catch (IException *e)
+                            {
+                                EXCLOG(e);
+                                e->Release();
+                                continue;
+                            }
+                            if (!locked)
+                            {
+                                part.lockProperties(30000);
+                                locked = true;
+                                PROGLOG("File '%s' has missing @size attributes", lfn);
+                            }
+                            PROGLOG("Part %u: Setting @size to %" I64F "u", p+1, partSize);
+                            attrs.setPropInt64("@size", partSize);
+                        }
+                        else
+                            PROGLOG("File '%s' missing @size on part %u", lfn, p+1);
+                    }
+                }
+            }
+        }
+        else
+            WARNLOG("File '%s' not found", lfn);
+    }
+    catch (IException *e)
+    {
+        EXCLOG(e);
+        e->Release();
+    }
+}
+
+void checkFileSize(IUserDescriptor *user, const char *lfnPattern, bool fix)
+{
+    if (containsWildcard(lfnPattern))
+    {
+        unsigned count = 0;
+        Owned<IDFAttributesIterator> iter = queryDistributedFileDirectory().getDFAttributesIterator(lfnPattern, user, true, false); // no supers
+        CCycleTimer timer;
+        if (iter->first())
+        {
+            while (true)
+            {
+                IPropertyTree &attr = iter->query();
+                const char *lfn = attr.queryProp("@name");
+                checkFileSizeOne(user, lfn, fix);
+                ++count;
+
+                if (!iter->next())
+                    break;
+                else if (timer.elapsedCycles() >= queryOneSecCycles()*10) // log every 10 secs
+                {
+                    PROGLOG("Processed %u files", count);
```

Comment:
Could this tracing become excessive (e.g. if run as an esp service)?

Reply:
this is never intended to be run as a service - it's very specific to the missing @size issue..
I think if it ever becomes something else, excessive tracing/other issues can be looked at then.